core/local/chokidar/initial_scan.js

```diff
@@ -0,0 +1,102 @@
+/* @flow */
+
+const chokidarEvent = require('./event')
+const logger = require('../../utils/logger')
+const metadata = require('../../metadata')
+
+const log = logger({
+  component: 'chokidar/initial_scan'
+})
+
+const NB_OF_DELETABLE_ELEMENT = 3
+
+/*::
+import type { ChokidarEvent } from './event'
+import type LocalEventBuffer from './event_buffer'
+import type Pouch from '../../pouch'
+
+export type InitialScan = {
+  ids: string[],
+  emptyDirRetryCount: number,
+  flushed: boolean,
+  resolve: () => void
+}
+
+export type InitialScanOpts = {
+  buffer: LocalEventBuffer<ChokidarEvent>,
+  initialScan: ?InitialScan,
+  pouch: Pouch
+}
+*/
+
+const detectOfflineUnlinkEvents = async (
+  initialScan /*: InitialScan */,
+  pouch /*: Pouch */
+) /*: Promise<{offlineEvents: Array<ChokidarEvent>, unappliedMoves: string[], emptySyncDir: boolean}> */ => {
+  // Try to detect removed files & folders
+  const events /*: Array<ChokidarEvent> */ = []
+  const docs = await pouch.byRecursivePathAsync('')
+  const inInitialScan = doc =>
+    initialScan.ids.indexOf(metadata.id(doc.path)) !== -1
+
+  // the Syncdir is empty error only occurs if there was some docs beforehand
+  let emptySyncDir = docs.length > NB_OF_DELETABLE_ELEMENT
+  let unappliedMoves = []
+
+  for (const doc of docs) {
+    if (inInitialScan(doc) || doc.trashed || doc.incompatibilities) {
```

Comment:
Do we need to check for incompatibilities in the Atom Watcher as well? Or maybe it's not even necessary here?